### PR TITLE
js to ts files

### PR DESCRIPTION
**Pull request by TVVaishnavi**
converting js to ts

### CODE REVIEW

File: src/service/signup.ts [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+interface UserData {
+    name: string;
+    email: string;
+    password: string;
+}
+
+const createUser = async (userData: UserData): Promise<any> => {
+    const { name, email, password } = userData;
+    const hashPassword: string = await bcrypt.hash(password, 10);
```
Comment (Rithikaram29) [inline src/service/signup.ts:12]: check for duplicate emails to see if the user already exists.

File: src/service/bus.ts [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+        bookedSeat,
+        inAC,
+        arrival,
+        departure,
+        stoppings,
+        arriveTime,
+        departureTime,
+        date
+    } = busdata;
+
```
Comment (Rithikaram29) [inline src/service/bus.ts:31]: check if the busNumber is already present and use it to handle error to avoid duplicate data

Comment (TVVaishnavi) [reply src/service/bus.ts:31]: I have created the condition in controller/bus so there won't be duplicated

File: src/service/signup.ts (hunk 1 above)
Comment (TVVaishnavi) [reply src/service/signup.ts:12]: I checked it and there won't be any duplicate emails.

**Pull request merged**